uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/models/HabitListTest.kt

```diff
@@ -0,0 +1,307 @@
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import java.io.IOException
+import java.io.StringWriter
+import java.util.ArrayList
+
+class HabitListTest : BaseUnitTest() {
+    @get:Rule
+    var thrown = ExpectedException.none()!!
+    private var habitsArray: ArrayList<Habit>? = null
+    private var activeHabits: HabitList? = null
+    private var reminderHabits: HabitList? = null
+    @Throws(Exception::class)
+    override fun setUp() {
+        super.setUp()
+        habitsArray = ArrayList()
+        for (i in 0..9) {
+            val habit = fixtures!!.createEmptyHabit()
+            habitList!!.add(habit)
+            habitsArray!!.add(habit)
+            if (i % 3 == 0) habit.reminder = Reminder(8, 30, WeekdayList.EVERY_DAY)
+        }
+        habitsArray!![0].isArchived = true
+        habitsArray!![1].isArchived = true
+        habitsArray!![4].isArchived = true
+        habitsArray!![7].isArchived = true
+        activeHabits = habitList!!.getFiltered(HabitMatcherBuilder().build())
+        reminderHabits = habitList!!.getFiltered(
+            HabitMatcherBuilder()
+                .setArchivedAllowed(true)
+                .setReminderRequired(true)
+                .build()
+        )
+    }
+
+    @Test
+    fun testSize() {
+        MatcherAssert.assertThat(habitList!!.size(), CoreMatchers.equalTo(10))
+        MatcherAssert.assertThat(activeHabits!!.size(), CoreMatchers.equalTo(6))
+        MatcherAssert.assertThat(reminderHabits!!.size(), CoreMatchers.equalTo(4))
+    }
+
+    @Test
+    fun testGetByPosition() {
+        MatcherAssert.assertThat(
+            habitList!!.getByPosition(0),
+            CoreMatchers.equalTo(
+                habitsArray!![0]
+            )
+        )
+        MatcherAssert.assertThat(
+            habitList!!.getByPosition(3),
+            CoreMatchers.equalTo(
+                habitsArray!![3]
+            )
+        )
+        MatcherAssert.assertThat(
+            habitList!!.getByPosition(9),
+            CoreMatchers.equalTo(
+                habitsArray!![9]
+            )
+        )
+        MatcherAssert.assertThat(
+            activeHabits!!.getByPosition(0),
+            CoreMatchers.equalTo(
+                habitsArray!![2]
+            )
+        )
+        MatcherAssert.assertThat(
+            reminderHabits!!.getByPosition(1),
+            CoreMatchers.equalTo(habitsArray!![3])
+        )
+    }
+
+    @Test
+    fun testGetById() {
+        val habit1 = habitsArray!![0]
+        val habit2 = habitList!!.getById(habit1.id!!)
+        MatcherAssert.assertThat(habit1, CoreMatchers.equalTo(habit2))
+    }
+
+    @Test
+    fun testGetById_withInvalidId() {
+        Assert.assertNull(habitList!!.getById(100L))
+    }
+
+    @Test
+    fun testOrdering() {
+        val h1 = fixtures!!.createEmptyHabit()
+        h1.name = "A Habit"
+        h1.color = PaletteColor(2)
+        h1.position = 1
```

Comment:
Could you use `apply` to make this look nicer?
```kotlin
val h1 = fixtures.createEmptyHabit().apply {
    name = "A Habit"
    color = [...]
}
```
An alternative would be making `name`, `color`, `position` etc arguments of `createEmptyHabit`, but provide default values.